### DISCUSSION TITLE
docker-compose inside a NodeJS Container

### DISCUSSION THREAD

**Discussion by JarodStreck** [PDMLab/docker-compose#221] [Q&A]
I was wondering if it works inside a Docker container using the host machine Docker API. 
Basically I have a container with a NodeJS app that manage containers. The issue is that docker-compose run the command in the shell but the container itself doesn't have docker nor docker-compose installed. Is there any work-around for that ?

**Comment by Steveb-p** [comment]
This library works by calling `docker-compose` binary, so anything that this binary can do, this can do as well.

However, without said binary, it will simply not work. You would either need to be able to call `docker-compose` inside the container you're working with, or have it spun a different container (docker-in-docker) and execute the commands there.

It might be possible to use `dockerode` library to work nowadays to execute `compose` commands, but I'm not familiar with docker API changes to be certain.

**Reply by JarodStreck** [reply]
Dockerode doesn't run docker-compose . I could actually install docker and docker-compose inside that container and create a new context and connect to my host machine with SSH to get the docker API. No idea if it would work, have to try it. 
I'm also very sad that I can't just specify to docker-compose where the docker engine is located.
